rest/routes.go

```diff
@@ -830,3 +825,44 @@ func (s *Service) createUserCert(usr string) error {
 
 	return nil
 }
+
+func (s *Service) fetchAndValidateCert(usr string) ([]byte, error) {
+	cert, err := depot.GetCertificate(s.Depot, usr)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem getting certificate")
+	}
+
+	rawCert, err := cert.GetRawCertificate()
+	if err != nil {
+		return nil, errors.Wrap(err, "problem getting raw certificate")
+	}
+	if rawCert.NotAfter.Before(time.Now().Add(time.Hour)) {
```

Comment:
this feels like it should be a constant, and after thinking let's renew if <4h and make certs last for 48 hours? We might want to push this up to 2weeks too? I'm worried about not having this configurable.